src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -812,8 +831,157 @@ def __set_errors_json(self, error_count_by_operation, errors_by_operation):
             "details": errors_by_operation,
             "message": message
         }
+
+    def __set_error_detail(self, error_code, message):
+        formatted_message = self.__ensure_error_message_restriction_compliance(message)
+        return {
+            "code": str(error_code),
+            "message": str(formatted_message)
+        }
     # endregion
 
+    # region - Patch Truncation
+    def log_truncated_packages(self):
+        """ log the removed packages from patches in CoreMain after main operation are marked completed """
```

Comment:
nit: """ log details of all the removed patches from status """

do not mention where this is called from,. A function, if globally available, is never restricted to being called from just one place and the function's implementation should not be designed based on where it's being called from.

Reply:
done